agents/lakeshore372/LS372_agent.py

```diff
@@ -258,21 +280,33 @@ def start_acq(self, session, params=None):
                         # Track the last channel we measured
                         previous_channel = self.module.get_active_channel()
 
+                    current_time = time.time()
                     data = {
-                        'timestamp': time.time(),
+                        'timestamp': current_time,
                         'block_name': active_channel.name,
                         'data': {}
                     }
 
                     # Collect both temperature and resistance values from each Channel
                     channel_str = active_channel.name.replace(' ', '_')
-                    data['data'][channel_str + '_T'] = \
-                        self.module.get_temp(unit='kelvin', chan=active_channel.channel_num)
-                    data['data'][channel_str + '_R'] = \
-                        self.module.get_temp(unit='ohms', chan=active_channel.channel_num)
+                    temp_reading = self.module.get_temp(unit='kelvin',
+                                                        chan=active_channel.channel_num)
+                    res_reading = self.module.get_temp(unit='ohms',
+                                                       chan=active_channel.channel_num)
+                    data['data'][channel_str + '_T'] = temp_reading
+                    data['data'][channel_str + '_R'] = res_reading
+
+                    field_dict = {channel_str: {"T": temp_reading,
+                                                "R": res_reading,
+                                                "timestamp": current_time}}
+
+                    data_cache.update(field_dict)
+                    session.data = {'fields': data_cache}
```

Comment:
I think this probably works, but since `session.data` is persistent between loops, you could also just initialize session.data with 
```session.data = {'fields': {}}```
and update with:
```
session.data['fields']['channel_str'] = {
    "T": temp_reading,
    "R": res_reading,
    "timestamp": current_time
}
```
instead of keeping a `data_cache`.